java/code/src/com/redhat/rhn/domain/formula/FormulaFactory.java

```diff
@@ -943,6 +950,12 @@ public static boolean hasMonitoringDataEnabled(ServerGroup group) {
     @SuppressWarnings("unchecked")
     private static boolean hasMonitoringDataEnabled(Map<String, Object> formData) {
         Map<String, Object> exporters = (Map<String, Object>) formData.get("exporters");
+        // If we cannot extract the exporters from the form data the form data file might
+        // come from an older version of the prometheus-exporters formula package and we will
+        // use the example file instead
+        if (exporters == null) {
+            exporters = (Map<String, Object>) FormulaFactory.getPillarExample(PROMETHEUS_EXPORTERS).get("exporters");
+        }
```

Comment:
This will change the function logic and return `true` even if exporters has been explicitly disabled in the old `formData`.

Reply:
You are right. Thinking about it I don't see a way to properly react to the exporters being null in this case. Any suggestions? I dislike using old logic just to get the data in case the file might be outdated.